core/src/main/java/com/linecorp/armeria/common/util/RateLimitingSampler.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2013-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * The rate-limited sampler allows you to choose an amount of traces to accept on a per-second
+ * interval. The minimum number is 0 and the max is 2,147,483,647 (max int).
+ *
+ * <p>For example, to allow 10 traces per second, you'd initialize the following:
+ * <pre>{@code
+ * tracingBuilder.sampler(RateLimitingSampler.create(10));
+ * }</pre>
+ *
+ * <h3>Appropriate Usage</h3>
+ *
+ * <p>If the rate is 10 or more traces per second, an attempt is made to distribute the accept
+ * decisions equally across the second. For example, if the rate is 100, 10 will pass every
+ * decisecond as opposed to bunching all pass decisions at the beginning of the second.
+ *
+ * <p>However, this sampler is insensitive to the trace ID and will operate correctly even if they are
+ * not perfectly random.
+ *
+ * <h3>Implementation</h3>
+ *
+ * <p>The implementation uses {@link System#nanoTime} and tracks how many yes decisions occur
+ * across a second window. When the rate is at least 10/s, the yes decisions are equally split over
+ * 10 deciseconds, allowing a roll-over of unused yes decisions up until the end of the second.
+ *
+ * <p>Forked from brave-core 5.6.9 at b8c00c594cbf75a33788d3dc990f94b9c6f41c01
+ */
+final class RateLimitingSampler implements Sampler {
+    static Sampler create(int tracesPerSecond) {
+        if (tracesPerSecond < 0) {
```

Comment:
nit: How about using `checkArgument(...)?

Reply:
Thank you for review. Fixed in a5fe184.